### ISSUE TITLE
Error around constraint when it appears the constraint is satisfied

### ISSUE THREAD

**Issue by SeanKilleen** [chocolatey/choco#3300]
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my problem.
- [X] I have verified this is not an issue for a specific package.
- [X] I have verified this issue is not security related.
- [X] I confirm I am using **official**, and **not** unofficial, or modified, Chocolatey products.

### What You Are Seeing?

It's possible this is related to #3297 in terms of incorrect version numbers being returned, as that could explain the issue (if chocolatey isn't checking my local installs to see if constraints are satisfied). But it seems separate enough that it merited another issue. We can close if it ends up being related.

----

When attempting to upgrade a package in this case `choco upgrade dotCover dotMemory dotPeek` (or any combination thereof), I see:

> Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'

Typically I'd think this was a package issue, but the first thing I did was to check my installed packages. Sure enough when running `choco list`, I see the `KB2919442` package is listed as installed and it is at version `1.0.20160915`, which should satisfy the constraint. 

As a result, none of these installs can complete

### What is Expected?

A successful installation because the constraint is satisfied.

### How Did You Get This To Happen?

1. I (at some point in the past) had a package install `KB2919442` as a dependency
2. I ran `choco upgrade dotCover dotMemory dotPeek`
3. The installation process fails with the error message above around version constraints
4. I check the version, of the package, which is still available and hasn't changed in some time.

### System Details

* Operating System: Windows 11
* Windows PowerShell version: Windows Powershell v5.1.22621.1778
* Chocolatey CLI Version: v2.2.0
* Chocolatey Licensed Extension version: N/A
* Chocolatey License type: None
* Terminal/Emulator: Windows Terminal 1.17.11461.0

### Installed Packages

```bash
7zip 23.1.0
7zip.install 23.1.0
azure-cli 2.51.0
azure-data-studio 1.45.1
chocolatey 2.2.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-fastanswers.extension 0.0.2.2
chocolatey-visualstudio.extension 1.11.0
chocolatey-windowsupdate.extension 1.0.5
choco-package-list-backup 2023.6.28
datagrip 2023.2.0
docker-desktop 4.22.0
dotCover 2023.1.4
dotMemory 2023.1.4
dotnet 7.0.9
DotNet4.6.1 4.6.1055.20170308
dotnet-6.0-sdk 6.0.412
dotnet-6.0-sdk-4xx 6.0.412
dotnet-7.0-runtime 7.0.9
dotnet-7.0-sdk 7.0.306
dotnet-7.0-sdk-3xx 7.0.306
dotnetfx 4.8.0.20220524
dotnet-runtime 7.0.9
dotnet-sdk 7.0.306
dotPeek 2023.1.4
Firefox 116.0.0
git 2.41.0
git.install 2.41.0
github-desktop 3.2.7
GoogleChrome 115.0.5790.171
graphviz 8.1.0
jetbrains-rider 2023.2.0
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
KB3063858 1.0.0
KB3118401 1.0.5
microsoft-teams 1.6.0.18681
microsoft-windows-terminal 1.17.11461
ncrunch-vs2022 4.16.4
netfx-4.7.2 4.7.2
notepadplusplus 8.5.4
notepadplusplus.install 8.5.4
notion 2.0.49
nvm 1.1.9
nvm.install 1.1.11
PDFXchangeEditor 10.0.1.371
poshgit 0.7.3.1
powershell-core 7.3.6
ReSharper 2023.1.4
resharper-platform 232.0.20230730.152252
slack 4.32.127
sysinternals 2023.7.26
terraform 1.5.5
treesizefree 4.6.3
vcredist140 14.36.32532
visualstudio2022professional 117.6.5
visualstudio2022-workload-azure 1.0.0
visualstudio2022-workload-data 1.0.0
visualstudio2022-workload-netweb 1.0.0
visualstudio-installer 2.0.3
vscode 1.80.1
vscode.install 1.80.1
windirstat 1.1.2.20161210
wsl2 2.0.0.20210721
```

### Output Log

```bash
`choco upgrade dotCover dotMemory dotPeek --debug --verbose` yields:


Chocolatey is running on Windows v 10.0.22621.0
Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
Command line: "C:\ProgramData\chocolatey\choco.exe" upgrade dotCover dotMemory dotPeek --debug --verbose
Received arguments: upgrade dotCover dotMemory dotPeek --debug --verbose
RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
Sending message 'PreRunMessage' out if there are subscribers...
[Pending] Removing all pending packages that should not be considered installed...
Performing validation checks.
Global Configuration Validation Checks:
 - Package Exit Code / Exit On Reboot = Checked
System State Validation Checks:
 Reboot Requirement Checks:
 - Pending Computer Rename = Checked
 - Pending Component Based Servicing = Checked
 - Pending Windows Auto Update = Checked
 - Pending File Rename Operations = Checked
 - Pending Windows Package Installer = Checked
 - Pending Windows Package Installer SysWow64 = Checked
Cache Folder Lockdown Checks:
 - Elevated State = Checked
 - Folder Exists = Checked
 - Folder lockdown = Checked
The source 'https://community.chocolatey.org/api/v2/' evaluated to a 'normal' source type

NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
Configuration: CommandName='upgrade'|
CacheLocation='C:\Users\SeanKilleen\AppData\Local\Temp\chocolatey'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://community.chocolatey.org/api/v2/'|SourceType='normal'|
ShowOnlineHelp='False'|Debug='True'|Verbose='True'|Trace='False'|
Force='False'|Noop='False'|HelpRequested='False'|
UnsuccessfulParsing='False'|RegularOutput='True'|QuietOutput='False'|
PromptForConfirmation='False'|DisableCompatibilityChecks='False'|
AcceptLicense='False'|AllowUnofficialBuild='False'|
Input='dotCover dotMemory dotPeek'|AllVersions='False'|
SkipPackageInstallProvider='False'|SkipHookScripts='False'|
PackageNames='dotCover;dotMemory;dotPeek'|Prerelease='False'|
ForceX86='False'|OverrideArguments='False'|NotSilent='False'|
ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
CacheExpirationInMinutes='30'|AllowDowngrade='False'|
ForceDependencies='False'|PinPackage='False'|
Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.22621.0'|
Information.PlatformName='Windows 10'|
Information.ChocolateyVersion='2.2.0.0'|
Information.ChocolateyProductVersion='2.2.0'|
Information.FullName='choco, Version=2.2.0.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='SeanKilleen'|
Information.UserDomainName='AzureAD'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|
Information.IsLicensedAssemblyLoaded='False'|
Information.LicenseType='Foss'|
Information.CurrentDirectory='C:\Users\SeanKilleen'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
ListCommand.LocalOnly='False'|ListCommand.IdOnly='False'|
ListCommand.IncludeRegistryPrograms='False'|ListCommand.PageSize='25'|
ListCommand.Exact='False'|ListCommand.ByIdOnly='False'|
ListCommand.ByTagOnly='False'|ListCommand.IdStartsWith='False'|
ListCommand.OrderByPopularity='False'|ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
ListCommand.ExplicitPageSize='False'|
ListCommand.ExplicitSource='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='Unknown'|
ApiKeyCommand.Command='Unknown'|PinCommand.Command='Unknown'|
OutdatedCommand.IgnorePinned='False'|
ExportCommand.IncludeVersionNumbers='False'|Proxy.BypassOnLocal='True'|
TemplateCommand.Command='unknown'|CacheCommand.Command='Unknown'|
CacheCommand.RemoveExpiredItemsOnly='False'|
_ Chocolatey:ChocolateyUpgradeCommand - Normal Run Mode _
Upgrading the following packages:
dotCover;dotMemory;dotPeek
By upgrading, you accept licenses for the packages.
Running list with the following filter = ''
--- Start of List ---
Resolving resource PackageSearchResource for source C:\ProgramData\chocolatey\lib
7zip 23.1.0
7zip.install 23.1.0
azure-cli 2.51.0
azure-data-studio 1.45.1
chocolatey 2.2.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-fastanswers.extension 0.0.2.2
chocolatey-visualstudio.extension 1.11.0
chocolatey-windowsupdate.extension 1.0.5
choco-package-list-backup 2023.6.28
datagrip 2023.2.0
docker-desktop 4.22.0
dotCover 2023.1.4
dotMemory 2023.1.4
dotnet 7.0.9
DotNet4.6.1 4.6.1055.20170308
dotnet-6.0-sdk 6.0.412
dotnet-6.0-sdk-4xx 6.0.412
dotnet-7.0-runtime 7.0.9
dotnet-7.0-sdk 7.0.306
dotnet-7.0-sdk-3xx 7.0.306
dotnetfx 4.8.0.20220524
dotnet-runtime 7.0.9
dotnet-sdk 7.0.306
dotPeek 2023.1.4
Firefox 116.0.0
git 2.41.0
git.install 2.41.0
github-desktop 3.2.7
GoogleChrome 115.0.5790.171
graphviz 8.1.0
jetbrains-rider 2023.2.0
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
KB3063858 1.0.0
KB3118401 1.0.5
microsoft-teams 1.6.0.18681
microsoft-windows-terminal 1.17.11461
ncrunch-vs2022 4.16.4
netfx-4.7.2 4.7.2
notepadplusplus 8.5.4
notepadplusplus.install 8.5.4
notion 2.0.49
nvm 1.1.9
nvm.install 1.1.11
PDFXchangeEditor 10.0.1.371
poshgit 0.7.3.1
powershell-core 7.3.6
ReSharper 2023.1.4
resharper-platform 232.0.20230730.152252
slack 4.32.127
sysinternals 2023.7.26
terraform 1.5.5
treesizefree 4.6.3
vcredist140 14.36.32532
visualstudio2022professional 117.6.5
visualstudio2022-workload-azure 1.0.0
visualstudio2022-workload-data 1.0.0
visualstudio2022-workload-netweb 1.0.0
visualstudio-installer 2.0.3
vscode 1.80.1
vscode.install 1.80.1
windirstat 1.1.2.20161210
wsl2 2.0.0.20210721
--- End of List ---
Resolving resource ListResource for source https://community.chocolatey.org/api/v2/
[NuGet]   CACHE https://community.chocolatey.org/api/v2/$metadata
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages()?$filter=(tolower(Id)%20eq%20'dotcover')%20and%20IsLatestVersion&semVerLevel=2.0.0

You have dotCover v2023.1.4 installed. Version 2023.2.0 is available based on your source(s).
Resolving resource DependencyInfoResource for source https://community.chocolatey.org/api/v2/
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='dotCover',Version='2023.2.0')
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='resharper-platform'&semVerLevel=2.0.0
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','107.0.20161110.114412-eap8'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','111.0.20171117.142630-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','191.0.20190326.124314-eap03'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','201.0.20200428.171125'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','211.0.20210316.83918-eap08'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','221.0.20220309.120359-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','231.0.20230208.84359-eap04'
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='dotnet4.6.1'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919355'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotMemory 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotPeek 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Re-attempting package dependency resolution using additional available package information...
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotMemory 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotPeek 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.
Running list with the following filter = ''
--- Start of List ---
7zip 23.1.0
7zip.install 23.1.0
azure-cli 2.51.0
azure-data-studio 1.45.1
chocolatey 2.2.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-fastanswers.extension 0.0.2.2
chocolatey-visualstudio.extension 1.11.0
chocolatey-windowsupdate.extension 1.0.5
choco-package-list-backup 2023.6.28
datagrip 2023.2.0
docker-desktop 4.22.0
dotCover 2023.1.4
dotMemory 2023.1.4
dotnet 7.0.9
DotNet4.6.1 4.6.1055.20170308
dotnet-6.0-sdk 6.0.412
dotnet-6.0-sdk-4xx 6.0.412
dotnet-7.0-runtime 7.0.9
dotnet-7.0-sdk 7.0.306
dotnet-7.0-sdk-3xx 7.0.306
dotnetfx 4.8.0.20220524
dotnet-runtime 7.0.9
dotnet-sdk 7.0.306
dotPeek 2023.1.4
Firefox 116.0.0
git 2.41.0
git.install 2.41.0
github-desktop 3.2.7
GoogleChrome 115.0.5790.171
graphviz 8.1.0
jetbrains-rider 2023.2.0
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
KB3063858 1.0.0
KB3118401 1.0.5
microsoft-teams 1.6.0.18681
microsoft-windows-terminal 1.17.11461
ncrunch-vs2022 4.16.4
netfx-4.7.2 4.7.2
notepadplusplus 8.5.4
notepadplusplus.install 8.5.4
notion 2.0.49
nvm 1.1.9
nvm.install 1.1.11
PDFXchangeEditor 10.0.1.371
poshgit 0.7.3.1
powershell-core 7.3.6
ReSharper 2023.1.4
resharper-platform 232.0.20230730.152252
slack 4.32.127
sysinternals 2023.7.26
terraform 1.5.5
treesizefree 4.6.3
vcredist140 14.36.32532
visualstudio2022professional 117.6.5
visualstudio2022-workload-azure 1.0.0
visualstudio2022-workload-data 1.0.0
visualstudio2022-workload-netweb 1.0.0
visualstudio-installer 2.0.3
vscode 1.80.1
vscode.install 1.80.1
windirstat 1.1.2.20161210
wsl2 2.0.0.20210721
--- End of List ---
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages()?$filter=(tolower(Id)%20eq%20'dotmemory')%20and%20IsLatestVersion&semVerLevel=2.0.0

You have dotMemory v2023.1.4 installed. Version 2023.2.0 is available based on your source(s).
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='dotMemory',Version='2023.2.0')
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='resharper-platform'&semVerLevel=2.0.0
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','107.0.20161110.114412-eap8'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','111.0.20171117.142630-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','191.0.20190326.124314-eap03'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','201.0.20200428.171125'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','211.0.20210316.83918-eap08'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','221.0.20220309.120359-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','231.0.20230208.84359-eap04'
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='dotnet4.6.1'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919355'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotCover 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotPeek 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Re-attempting package dependency resolution using additional available package information...
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotCover 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotPeek 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.
Running list with the following filter = ''
--- Start of List ---
7zip 23.1.0
7zip.install 23.1.0
azure-cli 2.51.0
azure-data-studio 1.45.1
chocolatey 2.2.0
chocolatey-compatibility.extension 1.0.0
chocolatey-core.extension 1.4.0
chocolatey-dotnetfx.extension 1.0.1
chocolatey-fastanswers.extension 0.0.2.2
chocolatey-visualstudio.extension 1.11.0
chocolatey-windowsupdate.extension 1.0.5
choco-package-list-backup 2023.6.28
datagrip 2023.2.0
docker-desktop 4.22.0
dotCover 2023.1.4
dotMemory 2023.1.4
dotnet 7.0.9
DotNet4.6.1 4.6.1055.20170308
dotnet-6.0-sdk 6.0.412
dotnet-6.0-sdk-4xx 6.0.412
dotnet-7.0-runtime 7.0.9
dotnet-7.0-sdk 7.0.306
dotnet-7.0-sdk-3xx 7.0.306
dotnetfx 4.8.0.20220524
dotnet-runtime 7.0.9
dotnet-sdk 7.0.306
dotPeek 2023.1.4
Firefox 116.0.0
git 2.41.0
git.install 2.41.0
github-desktop 3.2.7
GoogleChrome 115.0.5790.171
graphviz 8.1.0
jetbrains-rider 2023.2.0
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
KB3063858 1.0.0
KB3118401 1.0.5
microsoft-teams 1.6.0.18681
microsoft-windows-terminal 1.17.11461
ncrunch-vs2022 4.16.4
netfx-4.7.2 4.7.2
notepadplusplus 8.5.4
notepadplusplus.install 8.5.4
notion 2.0.49
nvm 1.1.9
nvm.install 1.1.11
PDFXchangeEditor 10.0.1.371
poshgit 0.7.3.1
powershell-core 7.3.6
ReSharper 2023.1.4
resharper-platform 232.0.20230730.152252
slack 4.32.127
sysinternals 2023.7.26
terraform 1.5.5
treesizefree 4.6.3
vcredist140 14.36.32532
visualstudio2022professional 117.6.5
visualstudio2022-workload-azure 1.0.0
visualstudio2022-workload-data 1.0.0
visualstudio2022-workload-netweb 1.0.0
visualstudio-installer 2.0.3
vscode 1.80.1
vscode.install 1.80.1
windirstat 1.1.2.20161210
wsl2 2.0.0.20210721
--- End of List ---
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages()?$filter=(tolower(Id)%20eq%20'dotpeek')%20and%20IsLatestVersion&semVerLevel=2.0.0

You have dotPeek v2023.1.4 installed. Version 2023.2.0 is available based on your source(s).
[NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='dotPeek',Version='2023.2.0')
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='resharper-platform'&semVerLevel=2.0.0
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','107.0.20161110.114412-eap8'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','111.0.20171117.142630-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','191.0.20190326.124314-eap03'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','201.0.20200428.171125'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','211.0.20210316.83918-eap08'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','221.0.20220309.120359-eap06'
[NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='resharper-platform'&$skiptoken='resharper-platform','231.0.20230208.84359-eap04'
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='dotnet4.6.1'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919355'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='KB2919442'&semVerLevel=2.0.0
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotCover 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotMemory 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Re-attempting package dependency resolution using additional available package information...
[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotCover 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotMemory 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'
Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.

Chocolatey upgraded 0/3 packages. 3 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - dotCover - Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.
 - dotMemory - Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.
 - dotPeek - Unable to resolve dependency 'KB2919442': Unable to find a version of 'KB2919442' that is compatible with 'KB2919355 1.0.20160915 constraint: KB2919442 (>= 1.0.20160915)'.
Sending message 'PostRunMessage' out if there are subscribers...
Exiting with 1
```
```


### Additional Context

_No response_

**Comment by pauby** [comment]
Have you gone through the docs, specifically [this one](https://docs.chocolatey.org/en-us/choco/troubleshooting/dependency-troubles)?

**Comment by SeanKilleen** [comment]
@pauby nope! Thanks -- I'd reviewed the issues and checked the docs but missed this one in particular which will probably lead me in the right direction. I'll keep this updated based on what I read there.

**Comment by TheCakeIsNaOH** [comment]
@SeanKilleen it looks like your issue is related to `[NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'dotCover 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'dotPeek 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)', 'ReSharper 2023.1.4 constraint: resharper-platform (= 231.0.20230713.103137)'`

**Comment by SeanKilleen** [comment]
Finally got a little time to try to dig into this further. 

I did see the message above but was unsure why it would prevent me from upgrading chocolatey itself and was confused by multiple kinds of errors showing up. The script claimed the constraint is around `resharper-platform` version `231.0.20230713.103137`.  However, [that dependency does exist](https://community.chocolatey.org/packages/resharper-platform/231.0.20230713.103137) and yet when I try to install it via ``, I see an error: `resharper-platform v232.0.20230730.152252 already installed. Forcing reinstall of version '232.0.20230730.152252'.` which then seemed to fail because it couldn't be found.

🎉 Ultimately I found my answer! 🎉 I think previously I had followed an instruction to delete a chocolatey folder so that it could be recreated and owned by the right account. It appears when I did that it screwed up some installs. Running `choco install dotCover dotMemory resharper resharper-platform -f` helped get those back into place.

Closing this now - thanks for the pointers to help get there.
